src/main/java/ru/skypro/homework/controller/AdsController.java

```diff
@@ -14,158 +17,146 @@
     @PatchMapping("/{adId}")
-
-    public ResponseEntity<Ad> updateAd(@PathVariable int adId, @RequestBody CreateOrUpdateAdDTO updateAd) {
-
-        return ResponseEntity.ok().body(adsService.editAdById(adId, updateAd));
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "403", description = "Forbidden"),
+            @ApiResponse(responseCode = "404", description = "Not Found")
+    })
+    public ResponseEntity<Ad> updateAd(@PathVariable int adId, @RequestBody CreateOrUpdateAd updateAd,
+                                       Authentication authentication) throws AccessException {
+        if (adsService.checkAccessToAd(adId, authentication.getName())) {
+            return ResponseEntity.ok().body(adsService.editAdById(adId, updateAd));
+        }
+        log.info("Отказано в доступе для " + authentication.getName());
+        return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
     }
 
-    @Operation(
-            operationId = "getCurrentUserAds",
-            summary = "Получение всех объявлений авторизованного пользователя",
-            tags = {"Объявления"},
-            responses = {
-                    @ApiResponse(responseCode = "200", description = "OK"),
-                    @ApiResponse(responseCode = "401", description = "Unauthorized"),
-            }
-    )
-    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Получение всех объявлений авторизованного пользователя")
     @GetMapping("/me")
-
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized")
+    })
     public ResponseEntity<AdsDTO> getCurrentUserAds(Authentication authentication) {
-        return ResponseEntity.ok().body(adsService.getAllAdsByUser(authentication));
+        return ResponseEntity.ok().body(adsService.getAllAdsByUser(authentication.getName()));
     }
-    @Operation(
-            operationId = "updateImage",
-            summary = "Обновление картинки объявления",
-            tags = {"Объявления"},
-            responses = {
-                    @ApiResponse(responseCode = "200", description = "OK"),
-                    @ApiResponse(responseCode = "401", description = "Unauthorized"),
-                    @ApiResponse(responseCode = "403", description = "Forbidden"),
-                    @ApiResponse(responseCode = "404", description = "Not Found")
-            }
-    )
-    @PreAuthorize("hasRole('USER') and @adsService.getAdById(#adId).email == authentication.principal.username")
-    @PatchMapping(value = "/{adId}/image", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
 
+    @Operation(summary = "Обновление картинки объявления")
+    @PreAuthorize("hasRole('ADMIN') and @adsService.getAdById(#adId).email == authentication.name")
```

Comment:
```suggestion
    @PreAuthorize("hasRole('ADMIN') and @adsService.getAdById(#adId).username == authentication.name")
```
потому что поле в сущности User называется username, а не email